app/controllers/orders_controller.rb

```diff
@@ -1,13 +1,24 @@
 class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
   def add_to_order
-    if Order.last
-      order = Order.last
-    else
-      order = Order.create
+	  if session[:order_id] == nil || Order.find(session[:order_id]).finalized == true
```

Comment:
Sweet. Thanks for the tip. We were looking for a solution like that but we did not figure it out